samples/testing/basic/cases_test.go

```diff
@@ -0,0 +1,28 @@
+package basic
+
+import "testing"
+
+func TestCases(t *testing.T) {
+	// You can create many test cases succinctly with an anonymous struct array
+	testCases := []struct {
+		input, expected string
+	}{
+		{"Hello", "Hello M8"},
+		{"Goodbye", "Goodbye M8"},
+		{"💥💥💥", "💥💥💥 M8"},
+		{"", " M8"},
+	}
+
+	for _, test := range testCases {
+		result := AddM8(test.input)
+		if result != test.expected {
+			// If one fails, log the failure and fail the test
+			t.Logf("wanted %s, got %s", test.expected, result)
+			t.Fail()
```

Comment:
Use https://golang.org/pkg/testing/#T.Errorf 
> Errorf is equivalent to Logf followed by Fail.

Reply:
Maybe also mention `Fatalf` ?

Reply:
I already cover `Fatalf` in `sample_test.go` (soon to be renamed to `basics_test.go`)